File: java-webflux/src/main/java/com/ederfmatos/javawebflux/gateway/CurrencyGateway.java
package com.ederfmatos.javawebflux.gateway;

import com.ederfmatos.javawebflux.enums.Currency;
import java.math.BigDecimal;
import java.util.Map;
import reactor.core.publisher.Mono;

public sealed interface CurrencyGateway permits DefaultCurrencyGateway {
    Mono<Map<Currency, BigDecimal>> getCurrencies();
}

File: kotlin-webflux/src/main/kotlin/com/ederfmatos/kotlinwebflux/gateway/CurrencyGateway.kt
package com.ederfmatos.kotlinwebflux.gateway

import com.ederfmatos.kotlinwebflux.enums.Currency
import reactor.core.publisher.Mono
import java.math.BigDecimal

interface CurrencyGateway {
    fun getCurrencies(): Mono<Map<Currency, BigDecimal>>
}